jerry-libc/include/assert.h

```diff
@@ -25,6 +25,11 @@ extern "C"
 #endif /* __cplusplus */
 
 #ifndef NDEBUG
+
+#ifndef __GNUC__
+#define __builtin_expect(expression, expected_value) (expression)
+#endif /* !__GNUC__ */
```

Comment:
I would prefer to replace the __builtin_expect() to unlikely() here. The project should not rely on __builtin_expect.